extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/HttpConfiguration.java

```diff
@@ -35,7 +35,7 @@
      * The HTTP host
      */
     @ConfigItem(defaultValue = "0.0.0.0")
-    public String host;
+    public Optional<String> host;
```

Comment:
Why switching this property as optional? It has a default value, so the optional will never be empty.

Reply:
So if you only want to use unix domain sockets you can set
`
quarkus.http.host=
`

And it will disable listening on `host:port`. Do you think it would be clearer to use a different property to disable this?

Reply:
maybe have remove the `Optional` and add this properties with these default?

```
quarkus.http.host.enabled=true
quarkus.http.host.domain-socket.enabled=false
```

Reply:
`quarkus.http.host.domain-socket` won't have a default value so it must be optional.

So then we'd only need
```
quarkus.http.host.enabled=true
```

But having `domain-socket` and `http` enabled differently is a bit strange.

Reply:
Another option is to give `domain-sockets` a default value `/var/run/io.quarkus.app.socket`. That would keep it consistent.

Reply:
@cescoffier  I went ahead and made these changes.